# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xorg/startwm.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

More flexible root application startup.

--- xinit/xinitrc.cpp.orig	Wed Apr 27 09:13:07 1994
+++ xinit/xinitrc.cpp	Thu Dec 28 15:23:15 2000
@@ -24,10 +24,10 @@
     xmodmap $usermodmap
 fi
 
-XCOMM start some nice programs
+XCOMM start the window manager
 
-twm &
-xclock -geometry 50x50-1+1 &
-xterm -geometry 80x50+494+51 &
-xterm -geometry 80x20+494-0 &
-exec xterm -geometry 80x66+0+0 -name login
+WINDOWMANAGER="${WINDOWMANAGER:-startkde}"
+[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+[ "$WINDOWMANAGER" != "twm"   ] && exec $WINDOWMANAGER
+xsetroot -solid darkcyan ; exec twm ; exec xterm
